pkg/pipelines/environments/environments.go

```diff
@@ -64,6 +65,17 @@ func (b *envBuilder) Service(env *config.Environment, svc *config.Service) error
 	if _, ok := b.files[envBindingPath]; !ok {
 		b.files[envBindingPath] = createRoleBinding(env, envBasePath, b.cicdEnv.Name, b.saName)
 	}
+	// Add trigger bindings only if source url exists
+	if b.cicdEnv != nil && svc.SourceURL != "" {
```

Comment:
I don't think we need to check for b.cicdEnv being nil here, because a few lines above (not showing in this diff) it returns `if b.cicdEnv == nil`.